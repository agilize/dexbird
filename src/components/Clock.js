import React from 'react';
import PropTypes from 'prop-types';

export const Clock = ({ time, className }) => (
  <div className={className}>{time.toLocaleTimeString()}</div>
);

Clock.propTypes = {
  time: PropTypes.objectOf(Date).isRequired,
};

export default Clock;
