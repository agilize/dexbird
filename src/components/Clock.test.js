import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Clock from './Clock';

describe('Clock', () => {
  const currentTime = new Date('Sat Nov 04 2017 11:56:37 GMT-0300 (BRT)');
  const wrapper = shallow(<Clock time={currentTime}></Clock>);

  test('render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render with the value 11:56:37 AM without class name', () => {
    const expected = <div>11:56:37 AM</div>;
    expect(wrapper).toContainReact(expected);
  });

  test('render with the class name', () => {
    const wrapper = shallow(<Clock time={currentTime} className="florentina"></Clock>);
    const expected = <div className="florentina">11:56:37 AM</div>;

    expect(wrapper).toContainReact(expected);
    expect(wrapper).toMatchSnapshot();
  });
});
