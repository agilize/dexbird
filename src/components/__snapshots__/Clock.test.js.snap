// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock render like snapshot 1`] = `
<div>
  11:56:37 AM
</div>
`;

exports[`Clock render with the class name 1`] = `
<div
  className="florentina"
>
  11:56:37 AM
</div>
`;
